app/src/main/java/org/sopt/and/presentation/signIn/SignInRoute.kt

```diff
@@ -0,0 +1,205 @@
+
+import android.widget.Toast
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusManager
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.LocalLifecycleOwner
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.flowWithLifecycle
+import org.sopt.and.R
+import org.sopt.and.core.designsystem.component.GrayTextField
+import org.sopt.and.core.designsystem.component.LoginHelpButton
+import org.sopt.and.core.designsystem.component.SNSLogin
+import org.sopt.and.core.designsystem.theme.BackgroundBlack
+import org.sopt.and.core.designsystem.theme.Gray60
+import org.sopt.and.core.designsystem.theme.MainBlue
+import org.sopt.and.core.designsystem.theme.pretendardFamily
+import org.sopt.and.presentation.signIn.SignInContract.SignInEffect
+import org.sopt.and.presentation.signIn.SignInContract.SignInEvent
+import org.sopt.and.presentation.signIn.SignInContract.SignInUiState
+
+@Composable
+fun SignInRoute(
+    modifier: Modifier = Modifier,
+    navigateToMy: (isLoginSuccess: Boolean) -> Unit,
+    navigateToSignUp: () -> Unit,
+    userViewModel: UserViewModel,
+    viewModel: SignInViewModel = hiltViewModel(),
+) {
+    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+    val snackbarHostState = remember { SnackbarHostState() }
+    val focusManager = LocalFocusManager.current
+    val context = LocalContext.current
+
+    val lifecycleOwner = LocalLifecycleOwner.current
+
+    LaunchedEffect(viewModel.sideEffect, lifecycleOwner) {
+        viewModel.sideEffect.flowWithLifecycle(lifecycle = lifecycleOwner.lifecycle)
```

Comment:
LaunchedEffect에 의해 자동으로 코루틴이 자동으로 취소되므로 flowWithLifecycle은 붙여주지 않아도 될 거 같아요!!
```suggestion
        viewModel.sideEffect.collect { ... }
```